news_trader/handlers/figi.py

```diff
@@ -16,6 +16,18 @@ def search_jobs(self, jobs: dict):
             "X-OPENFIGI-APIKEY": self._api_key,
         }
         response = requests.post(url=self._url, headers=headers, json=jobs)
+		
```

Comment:
I think this functionality can be implemented in more "clean" approach, without using `global`. 

Modify `__init__() `so that:

```
def __init__(self):
        self._api_key = os.environ.get("OPENFIGI_KEY")
        self._url = os.environ.get("OPENFIGI_URL")
        self._max_amount_of_requests_per_second = 5 if self._api_key is None else 20
```
        
        
Modify `search_jobs()` so that:

```
def search_jobs(self, jobs: dict):
      headers = {"Content-Type":  "text/json"}
      if self._api_key:
         headers.update({'X-OPENFIGI-APIKEY': self._api_key})
         
       ...# do not change the rest of this method
```

Modify` find_gm_tickers()` so that:

```
def find_gm_tickers(self, tickers) -> List[str]
   ...
   if iteration % self._max_amount_of_requests_per_second == 0:
          print('wait 60 secs')
          sleep(60)
```